### PR TITLE
Visual Studio Code language support

### LINKED ISSUE #5: we need a vscode extension
There is an LSP server (dada-lsp) and we even test it, but we don't have an extension. I have no idea how to write one. =) The vscode server is pretty bare-bones but it should support opening/closing files and reporting errors.

### PR DESCRIPTION
**Pull request by mbrobbel**
I'm still working on this but opening for visibility.
Closes #5.

### CODE REVIEW

File: editors/vscode/client/package.json [hunk 1]
```diff
@@ -0,0 +1,30 @@
+{
+	"name": "dada-language-server",
+	"description": "Dada Language Server",
+	"author": "Dada contributors",
+	"license": "MIT",
```
Comment (nikomatsakis) [inline editors/vscode/client/package.json:5]: should we make this "or apache"?

File: editors/vscode/language-configuration.json [hunk 2]
```diff
@@ -0,0 +1,104 @@
+{
+	"comments": {
+		"lineComment": "//",
```
Comment (nikomatsakis) [inline editors/vscode/language-configuration.json:3]: Nit: at least at present, comments are `#`, contrary to the tutorial.

**Review by nikomatsakis — CHANGES_REQUESTED** [review]
Cool! :)

File: editors/vscode/client/package.json (hunk 1 above)
Comment (mbrobbel) [reply editors/vscode/client/package.json:5]: Yes. I wrote this before https://github.com/dada-lang/dada/pull/101.

File: editors/vscode/language-configuration.json (hunk 2 above)
Comment (mbrobbel) [reply editors/vscode/language-configuration.json:3]: Yes. I just copy/pasted this file. The grammar uses `#`: https://github.com/dada-lang/dada/pull/99/files#diff-5131d9d0df4506ca526c37bc2d267def6694ca7ffbde11a41adf4e207c7ce37cR10-R12

**Comment by mbrobbel** [comment]
To run this locally:
- Add dada to your PATH e.g. `cargo install --path .`
- Make sure to install `node` + `npm`
- Install deps in editors/vscode: `npm install`
- Run the `npm` build task (editors/vscode)
- Start debug: "Launch Dada Language Server"
- Open a .dada file

Next steps:
- Setup nightly builds with binaray artifacts so that this extension can be extended to fetch a recent dada binary on startup
- Publish the extension (@nikomatsakis we can [automate this in CI](https://code.visualstudio.com/api/working-with-extensions/continuous-integration#github-actions-automated-publishing) - Can you setup the org for dada-lang?)
- Add some tests
- Fix the language server implementation. It does not clear diagnostics after doc syncs, and the offsets are incorrect

I can create follow-up issues for these taks, but looking to get some feedback first.

**Comment by nikomatsakis** [comment]
> Publish the extension (@nikomatsakis we can [automate this in CI](https://code.visualstudio.com/api/working-with-extensions/continuous-integration#github-actions-automated-publishing) - Can you setup the org for dada-lang?)

working on it

**Comment by nikomatsakis** [comment]
btw, I found some pretty serious bugs in the new entity salsa that I haven't gotten around to fixing, which led me to just recreate the state from scratch (i.e., create a new database) for the dada-web support. you probably need to do the same here.

**Comment by nikomatsakis** [comment]
@mbrobbel so I created a publisher (`dada-lang`).

**Pull request closed without merging**